src/OpenSSL/SSL.py

```diff
@@ -435,6 +450,10 @@ class _ALPNSelectHelper(_CallbackExceptionHelper):
     def __init__(self, callback):
         _CallbackExceptionHelper.__init__(self)
 
+        SSL_TLSEXT_ERR_OK = _lib.SSL_TLSEXT_ERR_OK
+        SSL_TLSEXT_ERR_NOACK = _lib.SSL_TLSEXT_ERR_NOACK
+        SSL_TLSEXT_ERR_ALERT_FATAL = _lib.SSL_TLSEXT_ERR_ALERT_FATAL
```

Comment:
Is there a reason we're mapping these to locals?

Reply:
I was scared about a reference cycle to `_lib` that passed through the C callback, which I assumed was the reason why the previous code returned 0 and 2 instead of these variables.

Reply:
Gosh, unless there's a specific failing test I'd much prefer to write normal code.

Absent some other evidence, the usual reason code is bad in pyOpenSSL is because it's old and bad, not because it was well thought out.